spec/registry.md

```diff
@@ -0,0 +1,156 @@
+            <option name="type" values="cardinal ordinal"/>
+            <option name="minimumIntegerDigits" pattern="positiveInteger"/>
+            <option name="minimumFractionDigits" pattern="positiveInteger"/>
+            <option name="maximumFractionDigits" pattern="positiveInteger"/>
+            <option name="minimumSignificantDigits" pattern="positiveInteger"/>
+            <option name="maximumSignificantDigits" pattern="positiveInteger"/>
+            <match values="one other"/>
+            <match pattern="anyNumber"/>
+        </matchSignature>
+
+        <formatSignature locales="en">
+            <input title="The number to format" pattern="anyNumber"/>
+            <option name="minimumIntegerDigits" pattern="positiveInteger"/>
+            <option name="minimumFractionDigits" pattern="positiveInteger"/>
+            <option name="maximumFractionDigits" pattern="positiveInteger"/>
+            <option name="minimumSignificantDigits" pattern="positiveInteger"/>
+            <option name="maximumSignificantDigits" pattern="positiveInteger"/>
+            <option name="style" readonly="true" values="decimal currency percent unit" default="decimal"/>
+            <option name="currency" readonly="true" pattern="currencyCode"/>
+        </formatSignature>
+    </function>
+</registry>
+```
+
+A localization engineer can then extend the registry by defining the following `customRegistry.xml` file.
+
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE registry SYSTEM "./registry.dtd">
+
+<registry>
+    <function name="noun">
+        <description>Handle the grammar of a noun.</description>
+        <formatSignature locales="en">
+            <input title="Noun id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="plural" values="one other"/>
+            <option name="case" values="nominative genitive" default="nominative"/>
+        </formatSignature>
+    </function>
+
+    <function name="adjective">
+        <description>Handle the grammar of an adjective.</description>
+        <formatSignature locales="en">
+            <input title="Adjective id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="plural" values="one other"/>
+            <option name="case" values="nominative genitive" default="nominative"/>
+        </formatSignature>
+        <formatSignature locales="en">
+            <input title="Adjective id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="accord"/>
+        </formatSignature>
+    </function>
+</registry>
+```
+
+Messages can now use the `:noun` and the `:adjective` functions.
+The following message references the first signature of `:adjective`,
+which expects the `plural` and `case` options:
+
+    {You see {$color :adjective article=indefinite plural=one case=nominative} {$object :noun case=nominative}!}
```

Comment:
What happens if you write `{You see {$color :adjective article=indefinite}}`? Since none of the options in the example are `required`, it's unclear which of the two signatures for `:adjective` should be used. Or does it not matter because the idea is that `:adjective` has a single implementation for all signatures?